services/comments.js

```diff
@@ -0,0 +1,277 @@
+        if (existingEntity) {
+            const entity = await strapi.query(reportModel.modelName, pluginName).create({
+                ...payload,
+                resolved: false,
+                related: id,
+            });
+            return sanitizeEntity(entity, { model: reportModel }) ;
+        }
+        throw new PluginError(409, 'Action on that entity is not allowed');
+    },
+
+    //
+    // Moderation
+    //
+
+    // Find single comment
+    findOneAndThread: async (id) => {
+        const { pluginName, service, model } = extractMeta(strapi.plugins);
+        const entity = await strapi.query( model.modelName, pluginName).findOne({ id }, ['threadOf', 'threadOf.reports', 'authorUser', 'related', 'reports']);
+        const entitiesOnSameLevel = await service.findAllInHierarchy(null, entity.threadOf ? entity.threadOf.id : null)
+        const selectedEntity = filterOurResolvedReports(sanitizeEntity(entity, { model }));
+        return {
+            selected: {
+                ...selectedEntity,
+                threadOf: selectedEntity.threadOf ? filterOurResolvedReports(selectedEntity.threadOf) : null,
+            },
+            level: entitiesOnSameLevel.map(_ => filterOurResolvedReports(sanitizeEntity(_, { model })))
+        };
+    },
+
+    // Block / Unblock a comment
+    blockComment: async (id) => {
+        const { pluginName, service, model } = extractMeta(strapi.plugins);
+        const existingEntity = await service.findOne(id);
+        const changedEntity = await strapi.query( model.modelName, pluginName).update(
+            { id },
+            { blocked: !existingEntity.blocked }
+        );
+        return sanitizeEntity(changedEntity, { model });
+    },
+
+    // Block / Unblock a comment thread
+    blockCommentThread: async (id) => {
+        const { pluginName, service, model } = extractMeta(strapi.plugins);
+        const existingEntity = await service.findOne(id);
+        const changedEntity = await strapi.query( model.modelName, pluginName).update(
+            { id },
+            { blockedThread: !existingEntity.blockedThread }
+        );
+        await service.blockCommentThreadNested(id, !existingEntity.blockedThread)
+        return sanitizeEntity(changedEntity, { model });
+    },
+
+    blockCommentThreadNested: async (id, blockStatus) => {
+        const { pluginName, service, model } = extractMeta(strapi.plugins);
+        try {
+            const changedEntities = await strapi.query(model.modelName, pluginName).update(
+                { threadOf: id },
+                { blockedThread: blockStatus }
+            );
+            if (changedEntities) {
+                const changedEntitiesList = changedEntities instanceof Array ? changedEntities : [changedEntities];
+                const nestedTransactions = await Promise.all(
+                    changedEntitiesList.map(async _ =>
+                        await service.blockCommentThreadNested(_.id, blockStatus)
```

Comment:
I think that usage of `await` inside of `map` will make this sequential despite usage of `Promise.all`